backend/logic/users/users.js

```diff
@@ -167,10 +169,76 @@ var getCompletedCourses = async studentID => {
   return courses;
 };
 
+var login = async () => {
+	
+	let LocalStrategy = require('passport-local').Strategy;
+		
+	//LOCAL SIGNIN
+	app.passport.use('local-signin', new LocalStrategy(  
+	{
+		// by default, local strategy uses username and password, we will override with email
+		usernameField : 'username',
+		passwordField : 'password',
+		passReqToCallback : true // allows us to pass back the entire request to the callback
+	},
+	
+	function(req, username, password, done) {
+		let error;
```

Comment:
you can refactor all the error checking logic into a separate function to clean this one up a bit